service/release.go

```diff
@@ -50,7 +50,14 @@ func (s *ReleaseService) Create(
 		return model.Release{}, apierrors.NewReleaseUnprocessableError().Wrap(err).WithMessage(err.Error())
 	}
 
-	// TODO check if release name is unique per project
+	unique, err := s.isReleaseTitleUniqueInProject(ctx, projectID, rls.ReleaseTitle)
+	if err != nil {
+		return model.Release{}, err
+	}
+	if !unique {
+		return model.Release{}, apierrors.NewReleaseDuplicateTitleError().Wrap(err)
+	}
+
 	if err := s.repo.Create(ctx, rls); err != nil {
```

Comment:
Can't we simply catch an appropriate SQL error here (smth like unique constraint violation).
Maybe already asked, but don't remember the reason.. 🤔
In such case, might deserve to be added as a comment...

Reply:
No, we can. I updated the PR and love the simplicity of the solution 👍

Originally, I thought you should always try to catch the error to avoid reaching the database if possible. However, I can see that this approach (at least in this case) creates more complexity and could result in a race condition, where a new record with the same name might be created in the meantime (after you check the name but before you insert a new record).

If you approve the solution I will also use the solution in the rest of the application.

Reply:
Yep. Repo returns API error. 👌